### DISCUSSION TITLE
SchmidTensor Function

### DISCUSSION THREAD

**Discussion by bibiboom89** [mtex-toolbox/mtex#1674] [Ask Anything] [answered]
Hi,

By definition the Schmid Tensor is
![Screenshot from 2022-12-01 19-01-36](https://user-images.githubusercontent.com/63070505/205184546-c0e8ba18-182c-4704-a2ef-837a315e8b5c.png)
where 'n^s' is the normal of the slip plan and 'l^s' the slip direction, equivalent to the Burger's vector.
So by definition, the Schmid Tensor should be symmetric.

When I try the following code:
```
n = rand(1,3); n = n / norm(n);
b = rand(1,3); b = b / norm(b);
test1 = SchmidTensor( vector3d(n),vector3d(b) )
test2 = (1/2)*(b'*n+n'*b)
```
I get the following results:
```
test1 = velocityGradientTensor (show methods, plot)
  rank: 2 (3 x 3)
 
 *10^-2
  26.9  5.36 55.35
  7.35  1.46 15.12
 33.45  6.66 68.82

test2 =

    0.2690    0.0635    0.4440
    0.0635    0.0146    0.1089
    0.4440    0.1089    0.6882
```

So my question is : did I do something wrong, am I missing something, or is the function SchmidTensor wrong ?

**Comment by Tijmenvermeij** [comment, marked as answer]
I don't know your definition of Schmid tensor... 

The Schmid tensor I'm familiar with should just describe the deformation gradient tensor, or displacement gradient tensor, of the slip system, which is just the dyadic product between the slip direction and slip plane normal (+ unity tensor for deformation gradient tensor). In your code, this would be b' * n

**Comment by bibiboom89** [comment]
Oh yeah ok, so to get back to my definition (in scientific article, this is called the orientation tensor, which is used to project the stress tensor on the slip system and calculate the resolved shear stress), I just need to do (test1.M+test1.M')/2.
Thank you for your answer.
Have a great day.
